client/rest-high-level/src/test/java/org/elasticsearch/client/core/AcknowledgedResponseResponseTests.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.core;
+
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+
+public class AcknowledgedResponseResponseTests extends AbstractXContentTestCase<AcknowledgedResponse> {
```

Comment:
Hmm, im not sure we need the toXContent in AcknowledgedResponse, can we make this test look like the other tests that use `AbstractXContentTestCase.xContentTester`, and remove the toXContent from the AcknowledgedResponse.

Also, I think this class name has one too many Response's in its name.

Reply:
> im not sure we need the toXContent in AcknowledgedResponse, can we make this test look like the other tests that use AbstractXContentTestCase.xContentTester, and remove the toXContent from the AcknowledgedResponse.

Agreed, there is no need for the response classes to serialize themselves. We just need them to be parsed from xcontent to java objects. I made this change but then realized that I then also need to change the `StartRollupJobResponse`, `PutRollupJobResponse` and `StartRollupJobResponse` response classes and their test. So I will do that in a follow up change if that is ok with you.